be/src/exec/pipeline/nljoin/nljoin_probe_operator.cpp

```diff
@@ -186,10 +186,53 @@ void NLJoinProbeOperator::_reset_build_chunk_index() {
     _move_build_chunk_index(0);
 }
 
-void NLJoinProbeOperator::_next_build_chunk_index() {
+void NLJoinProbeOperator::_next_build_chunk_index_for_other_join() {
     _move_build_chunk_index(_curr_build_chunk_index + 1);
 }
 
+// probe one chunk from right
+void NLJoinProbeOperator::_next_probe_row_index_for_inner_join() {
+    // next probe row
+    _probe_row_current++;
+    // probe chunk iter over
+    if (_probe_row_current >= _probe_chunk->num_rows()) {
+        // next build chunk
+        _curr_build_chunk_index++;
+        if (_curr_build_chunk_index >= _num_build_chunks()) {
+            // build chunks iter over
+            _curr_build_chunk_index = 0;
+            _probe_row_current = 0;
+            _probe_chunk = nullptr;
+        } else {
+            // next build chunk
+            _probe_row_current = 0;
+            _curr_build_chunk = _cross_join_context->get_build_chunk(_curr_build_chunk_index);
+            _build_row_current = 0;
+        }
```

Comment:
when to probe new tuple, the build side should restart from the begin, regardless the current build index?

Reply:
Yes, if the chunk size of build size is larger then probe size, always fetch one chunk from build size.